test/test_emulation/__init__.py

```diff
@@ -92,12 +92,19 @@ def udev_rules(self):
     def server(self):
         p = None
         try:
-            p = subprocess.Popen('python -m ratbag_emu 2&>1 >/dev/null', shell=True)
+            stdout = open('ratbag-emu-log-stdout.txt', 'w')
+            stderr = open('ratbag-emu-log-stderr.txt', 'w')
+
+            p = subprocess.Popen(['/usr/bin/env', 'python3', '-m', 'ratbag_emu'], stdout=stdout, stderr=stderr)
             sleep(2)
             yield
         finally:
             if p:
                 p.kill()
+            if stdout:
+                stdout.close()
+            if stderr:
+                stderr.close()
```

Comment:
is there any case where stdout or stderr can be None?

Reply:
Maybe? I am not sure.

Reply:
see https://github.com/libratbag/ratbag-emu/commit/6f6a21ed6947b7268b954bd4a1d78eaa9faec0a6 with a `with` statement you don't need to check and close the file, the context manager will do it for you.

Reply:
Sure.